iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/RegulatingPoint.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.commons.util.trove.TBooleanArrayList;
+import com.powsybl.iidm.network.StaticVarCompensator;
+import gnu.trove.list.array.TIntArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Supplier;
+
+/**
+ * @author Miora Vedelago <miora.ralambotiana at rte-france.com>
+ */
+class RegulatingPoint implements MultiVariantObject {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RegulatingPoint.class);
+
+    private final String regulatedEquipmentId;
+    private final Supplier<TerminalExt> localTerminalSupplier;
+    private TerminalExt terminal = null;
```

Comment:
As often in these cases, I'm getting lost in asking myself who's regulating / regulated! Maybe it would help if you rename this to `regulatingTerminal`?